cryptoconditions/crypto.py

```diff
@@ -1,269 +1,161 @@
-    def get_verifying_key(self):
-        """
-        Get the associated verifying key
+from cryptoconditions import exceptions
 
-        Returns:
-            A VerifyingKey object
-        """
 
-    @abstractmethod
-    def to_ascii(self, prefix, encoding):
-        """
-        Encode the external value
-
-        Args:
-            prefix:
-            encoding:
-        """
+class Base58Encoder(object):
 
     @staticmethod
-    @abstractmethod
-    def encode(private_value):
-        """
-        Encode the internal private_value to base58
-
-        Args:
-            private_value:
-        """
-
-    @staticmethod
-    @abstractmethod
-    def decode(private_base58):
-        """
-        Decode the base58 private value to internal value
-
-        Args:
-            private_base58 (base58):
-        """
-        raise NotImplementedError
-
-
-class VerifyingKey(metaclass=ABCMeta):
-
-    @abstractmethod
-    def verify(self, data, signature):
-        """
-        Check the if the signature matches the data and this verifyingkey
-
-        Args:
-            data:
-            signature:
-
-        Returns:
-            boolean:
-        """
-
-    @abstractmethod
-    def to_ascii(self, prefix, encoding):
-        """
-        Encode the external value
-
-        Args:
-            prefix:
-            encoding:
-        """
+    def encode(data):
+        return base58.b58encode(data).encode()
 
     @staticmethod
-    @abstractmethod
-    def encode(public_value):
-        """
-        Encode the public key to base58 represented by the internal values
-
-        Args:
-            public_value
-        """
-
-    @staticmethod
-    @abstractmethod
-    def decode(public_base58):
-        """
-        Decode the base58 public_value to internal value
-
-        Args:
-            public_base58 (base58):
-        """
-
-
-class Ed25519SigningKey(ed25519.SigningKey, SigningKey):
+    def decode(data):
+        return base58.b58decode(data)
+
+
+def _get_nacl_encoder(encoding):
+    if encoding == 'base58':
+        return Base58Encoder
+    elif encoding == 'base64':
+        return nacl.encoding.Base64Encoder
+    elif encoding == 'base32':
+        return nacl.encoding.Base32Encoder
+    elif encoding == 'base16':
+        return nacl.encoding.Base16Encoder
+    elif encoding == 'hex':
+        return nacl.encoding.HexEncoder
+    elif encoding is 'bytes':
+        return nacl.encoding.RawEncoder
+    else:
+        raise exceptions.UnknownEncodingError("Unknown or unsupported encoding")
+
+
+class Ed25519SigningKey(nacl.signing.SigningKey):
     """
     PrivateKey instance
     """
 
-    def __init__(self, key):
+    def __init__(self, key, encoding='base58'):
         """
-        Instantiate the private key with the private_value encoded in base58
+        Instantiate the private key with the private value.
 
         Args:
-            key (base58): base58 encoded private key
+            key (str): encoded private value.
```

Comment:
It works well we both. I guess the encoders handle both cases.
But to be consistent I will put bytes